lib/jsonapi/plugs/content_type_negotiation.ex

```diff
@@ -49,11 +49,18 @@ defmodule JSONAPI.ContentTypeNegotiation do
     {conn, content_type}
   end
 
+  defp respond({conn, @jsonapi, nil}) do
+    before_send(conn)
+  end
   defp respond({conn, @jsonapi, @jsonapi}) do
+    before_send(conn)
+  end
+  defp respond({conn, @jsonapi, _accepts}), do: send_error(conn, 406)
+  defp respond({conn, _content_type, _accepts}), do: send_error(conn, 415)
```

Comment:
I think this need to change.  The content_type isn't required but if it is provided and doesn't match `@jsonapi` we return 415.  Same applies for the accepts header, it can be blank.